sktime/transformations/base.py

```diff
@@ -269,6 +269,49 @@ def __radd__(self, other):
         else:
             return NotImplemented
 
+    def __getitem__(self, key):
+        """Magic [...] method, return column subsetted transformer.
+
+        First index does output subsetting, second index does input subsetting.
+
+        Keys must be valid inputs for `columns` in `ColumnSubset`.
+
+        Parameters
+        ----------
+        key: valid input for `columns` in `ColumnSubset`, or pair thereof
+            keys can also be a :-slice, in which case it is considered as not passed
+
+        Returns
+        -------
+        the following TransformerPipeline object:
+            ColumnSubset(columns2) * self * ColumnSubset(columns1)
+            where `columns1` is first item in `key` and `columns2` is second item
+            if only one of the two is passed, the corresponding pipeline item is skipped
+        """
+        from sktime.transformations.series.subset import ColumnSelect
+
+        def is_noneslice(obj):
+            res = isinstance(obj, slice)
+            res = res and obj.start is None and obj.stop is None and obj.step is None
+            return res
+
+        if isinstance(key, tuple):
+            if not len(key) == 2:
+                raise ValueError(
+                    "there should be one or two keys when calling [] or getitem, "
+                    "e.g., mytrafo[key], or mytrafo[key1, key2]"
+                )
+            columns1 = key[0]
+            columns2 = key[1]
+            if is_noneslice(columns1) and is_noneslice(columns2):
+                return self
+            elif is_noneslice(columns1):
```

Comment:
suggestion - it might also be worth supporting the t["a", : ] case, or at least catching it and passing an error?  Or potentially using this notation instead of t["a"]...  Reasoning -
- It keeps the pattern consistent across all use cases
- People might just try it, and *I think* strange things would happen with the current code?

Reply:
good point, added catch for that case

Reply:
(you should resolve the conversation once happy, or follow up if not)